components/eamxx/src/share/grid/remap/vertical_remapper.cpp

```diff
@@ -0,0 +1,286 @@
+#include "vertical_remapper.hpp"
+
+#include "share/grid/point_grid.hpp"
+#include "share/util/scream_vertical_interpolation.hpp"
+#include "share/io/scorpio_input.hpp"
+
+#include <ekat/kokkos/ekat_kokkos_utils.hpp>
+#include <ekat/ekat_pack_utils.hpp>
+#include <ekat/ekat_pack_kokkos.hpp>
+
+#include <numeric>
+
+namespace scream
+{
+VerticalRemapper::
+VerticalRemapper (const grid_ptr_type& src_grid,
+                  const std::string& map_file,
+                  const Field& lev_prof,
+                  const Field& ilev_prof)
+{
+  VerticalRemapper(src_grid,map_file,lev_prof,ilev_prof,-999999.0);
+}
+
+VerticalRemapper::
+VerticalRemapper (const grid_ptr_type& src_grid,
+                  const std::string& map_file,
+                  const Field& lev_prof,
+                  const Field& ilev_prof,
+                  const Real mask_val)
+ : AbstractRemapper()
+ , m_comm (src_grid->get_comm())
+ , m_mask_val(mask_val)
+{
+  using namespace ShortFieldTagsNames;
+
+  // Sanity checks
+  EKAT_REQUIRE_MSG (src_grid->type()==GridType::Point,
+      "Error! VerticalRemapper only works on PointGrid grids.\n"
+      "  - src grid name: " + src_grid->name() + "\n"
+      "  - src_grid_type: " + e2str(src_grid->type()) + "\n");
+  EKAT_REQUIRE_MSG (src_grid->is_unique(),
+      "Error! VerticalRemapper requires a unique source grid.\n");
+
+  // This is a vertical remapper. We only go in one direction
+  m_bwd_allowed = false;
+
+  // Gather the pressure level data for vertical remapping
+  set_pressure_levels(map_file);
+
+  // Create tgt_grid that is a clone of the src grid but with
+  // the correct number of level
+  auto tgt_grid = src_grid->clone("Point Grid",false);  //<-- ?? For Luca, why does the 1st arg have to be PointGrid?  Shouldn't the tgt_grid be of the same type as the src_grid if being cloned?  Note, choosing any other string threw an error for an "unsupported grid".
+  tgt_grid->reset_num_vertical_lev(m_num_remap_levs);
+  this->set_grids(src_grid,tgt_grid);
+
+  // Set the LEV and ILEV vertical profiles for interpolation from
+  register_vertical_source_field(lev_prof,"mid");
+  register_vertical_source_field(ilev_prof,"int");
+}
+
+VerticalRemapper::
```

Comment:
If the default destructor is fine, you should just do
```
~VerticalRemapper () = default;
```
in the header file. It makes your intent clear to whomever reads the header.